qiskit_machine_learning/algorithms/classifiers/qsvc.py

```diff
@@ -66,5 +74,54 @@ def quantum_kernel(self) -> QuantumKernel:
     @quantum_kernel.setter
     def quantum_kernel(self, quantum_kernel: QuantumKernel):
         """Sets quantum kernel"""
-        self._quantum_kernel = quantum_kernel
+        self._kernel_trainer = None
+
+        # If no qkernel passed, generate a default QuantumKernel
+        if not quantum_kernel:
+            backend = Aer.get_backend("qasm_simulator")
+            self._quantum_kernel = QuantumKernel(quantum_instance=backend)
+
+        # QuantumKernel passed
+        elif isinstance(quantum_kernel, QuantumKernel):
+            self._quantum_kernel = quantum_kernel
+
+            # If there are unbound user parameters, instantiate a QKTrainer
+            unbound_params = quantum_kernel.unbound_user_parameters()
+            if unbound_params:
+                # If user has not specified they wish to train the kernel,
+                # provide a warning
+                if not self.train_kernel:
+                    warnings.warn(
+                        f"""
+                        Unbound user parameters were detected in the quantum
+                        kernel feature map ({unbound_params}). If these are
+                        not bound before calling QSVC.fit, a computationally
+                        expensive kernel training procedure will precede
```

Comment:
kernel-training

Reply:
Does it make sense to spin up a default trainer rather than say error out and require the user to provide a trainer which they can customize etc. I guess if this default is generally useful then maybe this is fine to do it implicitly and warn rather than force the user to do it?